### DISCUSSION TITLE
Make autocompletion work via generics

### DISCUSSION THREAD

**Discussion by olivernybroe** [spatie/invade#4] [Ideas]
Currently the typehint just makes the IDE know it returns an Invader object, however we could make static analysis aware of what is possible by adding generic php docs to it. 🤔

**Comment by freekmurze** [comment]
Feel free to send a PR 👍

**Reply by olivernybroe** [reply]
#5 Opened a pr 👍  Apparently not as straightforward as I was hoping for 😅
